File: stack.py
def bracket_matching(input_string):
    input_array = []
    for bracket in input_string:
        if bracket == "(":
            input_array.append(")")
        elif bracket == "{":
            input_array.append("}")
        elif bracket == "[":
            input_array.append("]")
        else:
            if(not input_array or bracket != input_array[-1]):
                return False
            input_array.pop()
    if(not input_array):
        return True

if __name__ == '__main__':
    input_string = "{{[[(())]]}}"
    if(bracket_matching(input_string)):
        print "YES"
    else:
        print "NO"

File: stack.js
function isBracketMatching(inputSting) {
    let inputArray = [];
    for (let i = 0; i < inputSting.length; i++) {
        if (inputSting.charAt(i) === "(")
            inputArray.push(")");
        else if (inputSting.charAt(i) === "{")
            inputArray.push("}")
        else if (inputSting.charAt(i) === "[")
            inputArray.push("]")
        else {
            if (inputArray.length === 0 || inputSting.charAt(i) != inputArray[inputArray.length - 1])
                return false
            inputArray.pop();
        }
    }
    if (inputArray.length === 0)
        return true
}

var inputSting = "{{[[(())]]}]";

if (isBracketMatching(inputSting))
    console.log("YES")
else
    console.log("NO")